### PR TITLE
New package: hdf5-1.8.16

### PR DESCRIPTION
**Pull request by pulux**

### CODE REVIEW

File: srcpkgs/hdf5/template [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Template file for 'hdf5'
+pkgname=hdf5
+version=1.8.16
+revision=1
+build_style=gnu-configure
+only_for_archs="i686 x86_64 x86_64-musl"
```
Comment (xtraeme) [inline srcpkgs/hdf5/template]: what's the reason for `only_for_archs` ?

File: srcpkgs/hdf5/template [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+homepage="http://hdfgroup.org"
+distfiles="http://www.hdfgroup.org/ftp/HDF5/current/src/hdf5-${version}.tar.gz"
+checksum=ed17178abd9928a7237f30370189ba767b9e39e0db45917c2ac4665eb9cb4771
+
+post_install() {
+	vlicense COPYING
+}
+
+hdf5-devel_package() {
+	depends="zlib-devel"
```
Comment (xtraeme) [inline srcpkgs/hdf5/template]: `depends="${makedepends} ${sourcepkg}-${version}_${revision}"`

File: srcpkgs/hdf5/template (hunk 1 above)
Comment (pulux) [inline srcpkgs/hdf5/template]: on armx I get:
```
configure: error: cannot run test program while cross compiling
See `config.log' for more details
=> ERROR: hdf5-1.8.16_1: failed to run do_configure() at line 7.
```
the log says:
`#define PRINTF_LL_WIDTH "unknown"`

Comment (pullmoll) [inline srcpkgs/hdf5/template]: The switch to use in this case is `nocross=yes` then, because the package would probably compile on a native arm* build.

**Comment by xtraeme** [comment]
Please rebase to get this merged, thanks.

**Pull request merged**